From ee71d220e97ceeeda35794d6c1f9ca3a7ad23c04 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Thu, 21 Apr 2022 14:25:51 +0300
Subject: [PATCH 129/N] configure: Include a header in the check for
 _beginthread

Previously, the test tried compiling a call to the
_beginthread function without either declaring the function
(and its parameters) or including the corresponding header.

Since Clang 15 (which still is under development, so this may
still change before it's released) [1], implicit function
declarations are a hard error by default, when building code
in C99 mode (or newer).

[1] https://github.com/llvm/llvm-project/commit/7d644e1215b376ec5e915df9ea2eeb56e2d94626
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 209410c..9ccf893 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2239,7 +2239,7 @@ AC_MSG_RESULT([$with_nt_threads])
 if test $with_nt_threads = yes ; then
 AC_MSG_CHECKING([whether linking with nt-threads work])
 AC_LINK_IFELSE([
-    AC_LANG_PROGRAM([[]],[[_beginthread(0, 0, 0);]])
+    AC_LANG_PROGRAM([[#include <process.h>]],[[_beginthread(0, 0, 0);]])
   ],
   [AC_MSG_RESULT([yes])],
   [AC_MSG_ERROR([failed to link with nt-threads])])
